Products/Zuul/facades/modelqueryfacade.py

```diff
@@ -0,0 +1,78 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+from Products import Zuul
+from Products.Zuul.routers import TreeRouter
+from Products.Zuul.facades import ZuulFacade
+from Products.Zuul.catalog.interfaces import IModelCatalogTool
+from Products.Zuul.catalog.model_catalog_tool_helper import ModelCatalogToolHelper
+from Products.ZenUtils.Ext import DirectResponse
+
+
+
+class ModelQueryFacade(ZuulFacade):
+
+
+    def __init__(self, context):
+
+        self.context = context
+        super(ModelQueryFacade, self).__init__(context)
+        self.model_catalog = IModelCatalogTool(self.context)
+        self.model_catalog_helper = ModelCatalogToolHelper(self.model_catalog)
+        self.indexed, self.not_indexed = self.model_catalog.model_catalog_client.get_indexes()
+         
+
+    def filterIndexes(self, indexes):      
+
+        # Sensitive information which may be stored as a value for zProp in SOLR is not encrypted yet
+        # while that exclude it from the query.
+        filtered = ["zProperties"]
+        return [index for index in indexes if index not in filtered]
+
+
+    def getDevices(self, limit=200, globFilters=None, fields=None):
+
+        if fields is None:
+            indexes = self._getIndexes()
+        else:
+            indexes = self.filterIndexes(fields)
+
+        if 'uuid' not in indexes:
+            indexes.append('uuid')
+
+        # in case customer uses not indexed fields to filter results in SOLR remove them from the query
+        if globFilters:
+            for parameter in globFilters.keys():
+                if parameter not in self.indexed:
+                    globFilters.pop(parameter)
+
+        results = self.model_catalog_helper.get_devices(limit=limit, globFilters=globFilters, fields=indexes)
+        devices = [ b._data for b in results ]
+
+        if 'events' in indexes:
+            uuids = [dev['uuid'] for dev in devices]
+            zep = Zuul.getFacade('zep', self.context.dmd.primaryAq())
+            severities = zep.getEventSeverities(uuids)
+            for device in devices:
+                events = dict((zep.getSeverityName(sev).lower(), counts) for (sev, counts) in severities[device['uuid']].iteritems())
+                device['events'] = events
+
+        return devices, results.total
```

Comment:
Is this automatically serialized by the router?

Reply:
yes, it is, you can see demo in comments